include/etl/private/inplace_function_cpp11.h

```diff
@@ -0,0 +1,245 @@
+///\file
+
+/******************************************************************************
+The MIT License(MIT)
+
+Embedded Template Library.
+https://github.com/ETLCPP/etl
+https://www.etlcpp.com
+
+Copyright(c) 2025 BMW AG
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files(the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions :
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR rhs
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR rhsWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR rhs DEALINGS IN THE
+SOFTWARE.
+******************************************************************************/
+
+#ifndef ETL_INPLACE_FUNCTION_CPP11_INCLUDED
+#define ETL_INPLACE_FUNCTION_CPP11_INCLUDED
+
+#include "../platform.h"
+#include "../error_handler.h"
+#include "../exception.h"
+#include "../type_traits.h"
+#include "../utility.h"
+
+#ifndef ETL_INPLACE_FUNCTION_DEFAULT_CAPACITY
+  #define ETL_INPLACE_FUNCTION_DEFAULT_CAPACITY 32
```

Comment:
Can this be based on the target architecture? When building for 64 bit the bare minimum (`void()`) seems to be 40 bytes.